### ISSUE TITLE
"Cannot set readonly property: comparator for class: java.util.TreeMap" exception on assemble task

### ISSUE THREAD

**Issue by kivsiak** [KeepSafe/dexcount-gradle-plugin#11]
Here is a stacktrace:

```
Caused by: groovy.lang.ReadOnlyPropertyException: Cannot set readonly property: comparator for class: java.util.TreeMap
	at com.getkeepsafe.dexcount.PackageTree.addInternal(PackageTree.groovy:71)
	at com.getkeepsafe.dexcount.PackageTree.this$2$addInternal(PackageTree.groovy)
	at com.getkeepsafe.dexcount.PackageTree$this$2$addInternal$0.call(Unknown Source)
	at com.getkeepsafe.dexcount.PackageTree.addInternal(PackageTree.groovy:86)
	at com.getkeepsafe.dexcount.PackageTree.this$2$addInternal(PackageTree.groovy)
	at com.getkeepsafe.dexcount.PackageTree$this$2$addInternal$0.call(Unknown Source)
	at com.getkeepsafe.dexcount.PackageTree.addInternal(PackageTree.groovy:86)
	at com.getkeepsafe.dexcount.PackageTree.this$2$addInternal(PackageTree.groovy)
	at com.getkeepsafe.dexcount.PackageTree$this$2$addInternal$0.call(Unknown Source)
	at com.getkeepsafe.dexcount.PackageTree.addInternal(PackageTree.groovy:86)
	at com.getkeepsafe.dexcount.PackageTree.this$2$addInternal(PackageTree.groovy)
	at com.getkeepsafe.dexcount.PackageTree$this$2$addInternal$0.call(Unknown Source)
	at com.getkeepsafe.dexcount.PackageTree.addInternal(PackageTree.groovy:86)
	at com.getkeepsafe.dexcount.PackageTree.addMethodRef(PackageTree.groovy:59)
	at com.getkeepsafe.dexcount.PackageTree$addMethodRef.call(Unknown Source)
	at com.getkeepsafe.dexcount.DexMethodCountTask$_getPackageTree_closure3.doCall(DexMethodCountTask.groovy:112)
	at com.getkeepsafe.dexcount.DexMethodCountTask.getPackageTree(DexMethodCountTask.groovy:111)
	at com.getkeepsafe.dexcount.DexMethodCountTask_Decorated.getPackageTree(Unknown Source)
	at com.getkeepsafe.dexcount.DexMethodCountTask.countMethods(DexMethodCountTask.groovy:41)
	at com.getkeepsafe.dexcount.DexMethodCountTask$countMethods.call(Unknown Source)
	at com.getkeepsafe.dexcount.DexMethodCountPlugin$_applyAndroid_closure1_closure2_closure3.doCall(DexMethodCountPlugin.groovy:56)
	at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:548)
	at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:529)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 54 more

```

```
------------------------------------------------------------
Gradle 2.2.1
------------------------------------------------------------

Build time:   2014-11-24 09:45:35 UTC
Build number: none
Revision:     6fcb59c06f43a4e6b1bcb401f7686a8601a1fb4a

Groovy:       2.3.6
Ant:          Apache Ant(TM) version 1.9.3 compiled on December 23 2013
JVM:          1.8.0_25 (Oracle Corporation 25.25-b02)
OS:           Mac OS X 10.11 x86_64
```

**Comment by benjamin-bader** [comment]
Interesting; this seems to be a (mis)feature of Groovy, attempting to set a private final TreeMap field `comparator` outside of the map's constructor.  I haven't yet reproduced it, but I'll see what I can do to work around this.  Thanks for the report!

**Comment by benjamin-bader** [comment]
I haven't had any luck reproducing this crash.  Can you do me a favor?  What happens if you build your app with a newer Gradle wrapper version?

**Comment by benjamin-bader** [comment]
In the process of fixing issue #12, I may have fixed this.  If you still have problems with a newer Gradle version, try using the snapshot build I just uploaded: 'com.getkeepsafe.dexcount:dexcount-gradle-plugin:0.2.1-SNAPSHOT`

**Comment by kolipass** [comment]
Where can I take the snapshot?

http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.getkeepsafe.dexcount%22%20AND%20a%3A%22dexcount-gradle-plugin%22 Havent  'com.getkeepsafe.dexcount:dexcount-gradle-plugin:0.2.1-SNAPSHOT`

**Comment by benjamin-bader** [comment]
Apologies; the build is in Sonatype snapshots.  Your `build.gradle` will need to look something like this:

```groovy
buildscript {
  repositories {
    // other repositories e.g. jcenter() or mavenCentral()
    maven { url 'https://oss.sonatype.org/content/repositories/snapshots' }
  }

  dependencies {
    classpath 'com.getkeepsafe.dexcount:dexcount-gradle-plugin:0.2.1-SNAPSHOT'
  }
}
```

**Comment by kivsiak** [comment]
Snapshot version is working without any errors.

2015-09-11 21:12 GMT+02:00 Ben Bader <notifications@github.com>:

> Apologies; the build is in Sonatype snapshots. Your build.gradle will
> need to look something like this:
>
> buildscript {
>   repositories {
>     // other repositories e.g. jcenter() or mavenCentral()
>     maven { url 'https://oss.sonatype.org/content/repositories/snapshots' }
>   }
>
>   dependencies {
>     classpath 'com.getkeepsafe.dexcount:dexcount-gradle-plugin:0.2.1-SNAPSHOT'
>   }
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KeepSafe/dexcount-gradle-plugin/issues/11#issuecomment-139635390>
> .
>



-- 
Vladimir Abramov.

**Comment by benjamin-bader** [comment]
I released a stable version 0.2.1 containing this bugfix; it should propagate to Maven Central within a few hours.
